src/core/ddsc/src/dds_psmx.c

```diff
@@ -249,10 +266,54 @@ static dds_return_t psmx_instance_load (const struct ddsi_domaingv *gv, struct d
     goto err_init;
   }
   psmx_instance->priority = config->priority.value;
+
+  if (config->only_for_topics.size > 0 && config->forbidden_topics.size > 0) {
```

Comment:
I tend to think that having both allow and deny lists makes sense, then "it" (in this case, Iceoryx or PSMX, but it applies more generally) only gets used if allowed and not disallowed. In short, I think it would not consider this conflicting options.

Reply:
@eboasson The problem with having both options at the same time is how this interacts with the default state. If you have a topic that is on neither list, what do you do?
If the default is to use the PSMX then the allowlist makes no sense. If the default is not to use the PSMX then the denylist makes no sense.

So what this PR does is to set the default based on which list is. used. If an allowlist is provided then the default is "no". If a denylist is provided then the default is "yes". And that's why these two options are conflicting.